pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -129,3 +131,150 @@ func (nameScoping *DefaultNameScopingImpl) Handle(command utils.CommandEnum, clu
+	links := make([]string, 0)
+	for _, link := range containerConfig.HostConfig.Links {
+		linkSplit := strings.SplitN(link, ":", 2)
+		links = append(links, strings.TrimSpace(linkSplit[0]))
+	}
+
+	var m map[string]string
+	var err error
+	containerRefs := make([]string, 0)
+	containerRefs = append(containerRefs, affinityRef...)
+	containerRefs = append(containerRefs, volumesFrom...)
+	containerRefs = append(containerRefs, links...)
+	if m, err = getIDsFromContainerReferences(cluster, tenantId, containerRefs, affinityLabelFirst); err != nil {
+		return err
+	}
+	// update containerConfig
+	// update VolumesFrom
+	for i, k := range containerConfig.HostConfig.VolumesFrom {
+		containerConfig.HostConfig.VolumesFrom[i] = m[k]
+	}
+	// update affinity
+	if len(affinityRef) > 0 {
+		containerConfig.Env[affinityIndex] = "affinity:container==" + m[affinityRef[0]]
+	}
+	// update links
+	links = make([]string, 0)
+	linkSet := make(map[string]bool)
+	for _, link := range containerConfig.HostConfig.Links {
+		linkSplit := strings.SplitN(link, ":", 2)
+		containerIdName := strings.TrimSpace(linkSplit[0])
+		containerId := m[containerIdName]
+		if containerId != containerIdName {
+			linkSet[containerId+":"+containerIdName] = true
+		}
+		if len(linkSplit) > 1 {
+			linkSet[containerId+":"+strings.TrimSpace(linkSplit[1])] = true
+		}
+		for k := range linkSet {
+			links = append(links, k)
+		}
+	}
+	containerConfig.HostConfig.Links = links
+	return nil
+
+}
+func getIDsFromContainerReferences(cluster cluster.Cluster, tenantId string, containerReferences []string, affinityLabelFirst bool) (map[string]string, error) {
+	// containerReferences is a array of container references of the form long id, short id, or name
+	// if affinityLabelFirst is true the first element of containerReferences an affinity label reference of the form label:value
+	m := make(map[string]string)
+	for _, containerReference := range containerReferences {
+		m[containerReference] = ""
+	}
+	for _, container := range cluster.Containers() {
+		if container.Labels[headers.TenancyLabel] == tenantId {
+			for i, containerReference := range containerReferences {
+				var err error
+				var containerId string
+				if affinityLabelFirst && i == 0 {
+					containerId, err = getIDFromContainerLabel(container, tenantId, containerReference)
+				} else {
+					containerId, err = getContainerId(container, tenantId, containerReference)
+				}
+				if err == nil {
+					m[containerReference] = containerId
+					break
+				}
+			}
+		}
+	}
+	for _, containerReference := range containerReferences {
+		if m[containerReference] == "" {
+			err := fmt.Errorf(NOTAUTHORIZED_ERROR, containerReference)
+			return nil, err
+		}
+	}
+	return m, nil
+}
+
+func getIDFromContainerLabel(container *cluster.Container, tenantId string, affinityLabelValue string) (string, error) {
+	if container.Labels[headers.TenancyLabel] != tenantId {
+		return "", errors.New("not authorized to access this container.")
```

Comment:
@ezrasilvera    We can if you want, but the user never sees these "errors".  These errors are handled internally and simple means that the calling routine should continue its search. We could just as easily used errors.New("").  BTW: Originally it return "" but you asked for an error.